### ISSUE TITLE
firefox 57 doesn

### ISSUE THREAD

**Issue by xubinxian** [ChihChengYang/wfs.js#9]

**Comment by ChihChengYang** [comment]
Yeah, that was a known issue, Chrome seems to run more normal than Firefox. That must look into what  different behavior between browsers.

**Comment by weibinzhu** [comment]
I have figured it out.  

If you delete the firefox specific code, which make video restart form beginning, in onSBUpdateEnd, you will find out that video might stop when the network is slow. Normally video should resume automatically after a while when sufficient data are buffered. However in firefox, it just stop permanently.

If you look closer, you will also find that when this happen, the currentTime property still updates normally and  the 'pause' button still here. When you look at the GPU, you will notice an abnormally high GPU consumption, 100% in my computer, much higher than normal case.

So I think this might relate to firefox's hardware acceleration then I turned it off and you know what? problem solved.

If you really want to use hardware acceleration you can apply some cache buffer, in my case I changed the code in onH264DataParsing to: 
```js
this._parseAVCTrack(event.data);
      if (this.browserType === 1) {
        // Firefox
        if (this._avcTrack.samples.length >= 3) {
          this.remuxer.pushVideo(0, this.sn, this._avcTrack, this.timeOffset, this.contiguous);
          this.sn += 1;
        }
      } else {
        if (this._avcTrack.samples.length >= 20) {
          this.remuxer.pushVideo(0, this.sn, this._avcTrack, this.timeOffset, this.contiguous);
          this.sn += 1;
        }
      }
```

**Comment by jeffccy** [comment]
Thanks for the update on the issue.You may submit a pull request. :)
